src/lib/containers/FilePreview/EntityPreview.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import { useResolveLinkEntity } from '../../utils/functions/ResolveLinkEntity'
+import useGetEntityBundle from '../../utils/hooks/SynapseAPI/entity/useEntityBundle'
+import { ALL_ENTITY_BUNDLE_FIELDS } from '../../utils/SynapseConstants'
+import { FileEntity, isFileEntity } from '../../utils/synapseTypes'
+import { SynapseSpinner } from '../LoadingScreen'
+import FileEntityPreview from './FileEntityPreview'
+
+export type EntityPreviewProps = {
+  entityId: string
```

Comment:
Could be a "Reference" if you want to reuse the backend data model object

Reply:
Then again, if this entityId could have the dot notation then this is not a good fit.